src/Lazy/LList.elm

```diff
@@ -333,8 +333,44 @@ This function is performed lazily.
 
 -}
 sortWith : (a -> a -> Order) -> LList a -> LList a
-sortWith predicate =
-    Lazy.map (List.sortWith predicate)
+sortWith compare =
+    Lazy.map (List.sortWith compare)
+
+
+{-| Stable sort with for `LList`.
+The original order is guaranteed to be kept if the comparison returns `EQ`.
+
+    compareAge : Person -> Person -> Order
+    compareAge a b =
+        Basics.compare a.age b.age
+
+    llist [ { name = "Joe", age = 25 }, { name = "Sue", age = 25 }, { name = "Johann", age = 23 } ]
+        |> stableSortWith compareAge
+        |> toList
+    --> [ { name = "Johann", age = 23 }, { name = "Joe", age = 25 }, { name = "Sue", age = 25 } ]
+
+This function is performed lazily.
+
+-}
+stableSortWith : (a -> a -> Order) -> LList a -> LList a
+stableSortWith compare =
+    let
+        addIndex =
+            List.indexedMap (\i a -> ( a, i ))
+
+        compareWithIndex ( a1, i1 ) ( a2, i2 ) =
+            let
+                result =
+                    compare a1 a2
+            in
+            case result of
+                Basics.EQ ->
+                    Basics.compare i1 i2
+
+                _ ->
+                    result
+    in
+    Lazy.map (addIndex >> List.sortWith compareWithIndex >> List.map Tuple.first)
```

Comment:
hmm I did not know stable sort in elm has to do extra 2 folds over the list but I see list-extra does the same.